### PR TITLE
prep-v0.1.0-release

### PR DESCRIPTION
**Pull request by ahuang11**
https://github.com/PrefectHQ/prefect-docker/issues/27

### Example

Successfully ran with all the tasks in a single flow
```
from prefect import flow
from prefect_docker.images import pull_docker_image
from prefect_docker.containers import (
    create_docker_container,
    start_docker_container,
    get_docker_container_logs,
    stop_docker_container,
    remove_docker_container,
)


@flow
def test_docker_container_flow():
    pull_docker_image("prefecthq/prefect", "latest")
    container = create_docker_container(
        image="prefecthq/prefect", command="echo 'hello world!' && sleep 60"
    )
    start_docker_container(container_id=container.id)
    logs = get_docker_container_logs(container_id=container.id)
    print(logs)
    stop_docker_container(container_id=container.id)
    remove_docker_container(container_id=container.id)
    print("all done")
    return container
```

Outputs:
```
20:03:48.806 | INFO    | prefect.engine - Created flow run 'remarkable-squirrel' for flow 'test-docker-container-flow'
20:03:49.714 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'pull_docker_image-f9320424-0' for task 'pull_docker_image'
20:03:49.715 | INFO    | Flow run 'remarkable-squirrel' - Executing 'pull_docker_image-f9320424-0' immediately...
20:03:50.073 | INFO    | Task run 'pull_docker_image-f9320424-0' - Pulling image: prefecthq/prefect:latest.
20:04:11.859 | INFO    | Task run 'pull_docker_image-f9320424-0' - Finished in state Completed()
20:04:11.986 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'create_docker_container-ac4f93c1-0' for task 'create_docker_container'
20:04:11.986 | INFO    | Flow run 'remarkable-squirrel' - Executing 'create_docker_container-ac4f93c1-0' immediately...
20:04:12.491 | INFO    | Task run 'create_docker_container-ac4f93c1-0' - Creating container with 'prefecthq/prefect' image.
20:04:13.112 | INFO    | Task run 'create_docker_container-ac4f93c1-0' - Finished in state Completed()
20:04:13.282 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'start_docker_container-e92f6c4a-0' for task 'start_docker_container'
20:04:13.282 | INFO    | Flow run 'remarkable-squirrel' - Executing 'start_docker_container-e92f6c4a-0' immediately...
20:04:13.628 | INFO    | Task run 'start_docker_container-e92f6c4a-0' - Starting container 'b5a4e98aec4d3783aefb5e81c0d214f0ea2db1880b2c9434b953dc1cfdbbe8b8'.
20:04:14.109 | INFO    | Task run 'start_docker_container-e92f6c4a-0' - Finished in state Completed()
20:04:14.257 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'get_docker_container_logs-8be32a2f-0' for task 'get_docker_container_logs'
20:04:14.259 | INFO    | Flow run 'remarkable-squirrel' - Executing 'get_docker_container_logs-8be32a2f-0' immediately...
20:04:14.613 | INFO    | Task run 'get_docker_container_logs-8be32a2f-0' - Retrieving logs from 'b5a4e98aec4d3783aefb5e81c0d214f0ea2db1880b2c9434b953dc1cfdbbe8b8' container.
20:04:14.756 | INFO    | Task run 'get_docker_container_logs-8be32a2f-0' - Finished in state Completed()
20:04:14.892 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'stop_docker_container-aeb280c8-0' for task 'stop_docker_container'
20:04:14.893 | INFO    | Flow run 'remarkable-squirrel' - Executing 'stop_docker_container-aeb280c8-0' immediately...
hello world! && sleep 60

20:04:15.318 | INFO    | Task run 'stop_docker_container-aeb280c8-0' - Stopping container 'b5a4e98aec4d3783aefb5e81c0d214f0ea2db1880b2c9434b953dc1cfdbbe8b8'.
20:04:15.498 | INFO    | Task run 'stop_docker_container-aeb280c8-0' - Finished in state Completed()
20:04:15.622 | INFO    | Flow run 'remarkable-squirrel' - Created task run 'remove_docker_container-73ef219d-0' for task 'remove_docker_container'
20:04:15.623 | INFO    | Flow run 'remarkable-squirrel' - Executing 'remove_docker_container-73ef219d-0' immediately...
20:04:16.003 | INFO    | Task run 'remove_docker_container-73ef219d-0' - Removing container 'b5a4e98aec4d3783aefb5e81c0d214f0ea2db1880b2c9434b953dc1cfdbbe8b8'.
20:04:16.140 | INFO    | Task run 'remove_docker_container-73ef219d-0' - Finished in state Completed()
20:04:16.265 | INFO    | Flow run 'remarkable-squirrel' - Finished in state Completed()
all done
<Container: b5a4e98aec4d>
```

### Checklist

- [ ] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-docker/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-docker/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by ahuang11** [comment]
Docs looks good
image

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,18 +21,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## 0.1.0
 
-Released on ????? ?th, 20??.
+Released on October 21th, 2022.
```
Comment (desertaxle) [inline CHANGELOG.md:24]:
```suggestion
Released on October 21st, 2022.
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
A couple of things that I think need to be addressed before release:
- [ ] Add Credentials section to docs
- [ ] Add examples for `DockerRegistryCredentials` and `DockerHost`
- [ ] Document `DockerRegistryCredentials.login` arguments

I think it'd also be rad to add the example flow in your PR description to be the first example in the README.

File: README.md [hunk 2]
```diff
@@ -50,21 +50,36 @@ prefect block register -m prefect_docker
 
 Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://orion-docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://orion-docs.prefect.io/ui/blocks/).
 
-### Create a Docker container
+### Pull, create, start, log, stop, and remove Docker container
```
Comment (ahuang11) [inline README.md:53]:
```suggestion
### Pull image, and create, start, log, stop, and remove Docker container
```

**Comment by ahuang11** [comment]
Thanks for catching those!

File: prefect_docker/credentials.py [hunk 3]
```diff
@@ -9,6 +9,24 @@
...
+        @flow
+        def docker_registry_login_flow():
+            docker_host = DockerHost()
+            docker_registry_credentials = DockerRegistryCredentials(
+                username="my_username",
+                password="my_password",
+                registry_url="registry.hub.docker.com",
+            )
+            with docker_host.get_client() as client:
+                docker_registry_credentials.login(client)
```
Comment (desertaxle) [inline prefect_docker/credentials.py:28]: This doesn't need to be in a flow, so the minimal example might be more clear if we don't run it in a flow.
```suggestion
        from prefect_docker import DockerHost, DockerRegistryCredentials

        docker_host = DockerHost()
        docker_registry_credentials = DockerRegistryCredentials(
            username="my_username",
            password="my_password",
            registry_url="registry.hub.docker.com",
        )
        with docker_host.get_client() as client:
            docker_registry_credentials.login(client)
```

File: prefect_docker/host.py [hunk 4]
```diff
@@ -39,6 +39,22 @@ class DockerHost(Block):
...
+        from prefect_docker import DockerHost
+
+        @flow
+        def docker_host_get_client_flow():
+            docker_host = DockerHost(
+                base_url="tcp://127.0.0.1:1234",
+                max_pool_size=4
+            )
+            with docker_host.get_client() as client:
+                pass
```
Comment (desertaxle) [inline prefect_docker/host.py:56]: Similar idea that the flow isn't strictly necessary
```suggestion
        from prefect_docker import DockerHost

        docker_host = DockerHost(
        base_url="tcp://127.0.0.1:1234",
            max_pool_size=4
        )
        with docker_host.get_client() as client:
            ... # Use the client for Docker operations
```

**Review by desertaxle — COMMENTED** [review]
Couple of suggestions for the client examples

**Review by desertaxle — APPROVED** [review]
🚢 it!

**Pull request merged**